### PR TITLE
Progress tracker added to exercise editor along with progress bar

### PR DESCRIPTION
**Pull request by roryhmartin**
Numerical Student Progress tracker added to footer of exercise editor. 
Visual Progress Tracker added alongside breadcrumbs in footer

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -1,5 +1,5 @@
 MYSQL_HOST=db
-MYSQL_DATABASE=phpschool
+MYSQL_DATABASE=phpredisschool
```
Comment (AydinHassan) [inline .env.dist]:
```suggestion
MYSQL_DATABASE=phpschool
```

File: assets/components/ExerciseEditor.vue [hunk 2]
```diff
@@ -34,6 +34,8 @@ export default {
     workshop: Object,
     exercise: Object,
     student: Object,
+    totalExercises: Number,
+
```
Comment (AydinHassan) [inline assets/components/ExerciseEditor.vue]:
```suggestion
    totalExercises: Number
```

File: assets/components/ExerciseEditor.vue [hunk 3]
```diff
@@ -59,7 +61,9 @@ export default {
       activeTab: 0,
       newDependency: '',
       composerDeps: [],
-      loadingComposerAdd: false
+      loadingComposerAdd: false,
+      percentComplete: (this.student.state.total_completed / this.totalExercises) * 100,
+      totalCompleted:  this.student.state.total_completed,
```
Comment (AydinHassan) [inline assets/components/ExerciseEditor.vue]:
```suggestion
      totalCompleted: this.student.state.total_completed,
```

File: assets/components/ExerciseEditor.vue [hunk 4]
```diff
@@ -183,7 +187,7 @@ export default {
               :files="studentFiles"
               :file-select-function="studentSelectFile"
               :initial-selected-item="studentFiles[0]"
-              show-controls/>
+              show-controls/>
```
Comment (AydinHassan) [inline assets/components/ExerciseEditor.vue]:
```suggestion
              show-controls/>
```

File: assets/components/ExerciseEditor.vue [hunk 5]
```diff
@@ -209,7 +213,7 @@ export default {
               <div class="space-y-3">
                 <div class="grid grid-cols-3 gap-4">
                   <div class="h-2 bg-slate-700 rounded col-span-2"></div>
-                  <div class="h-2 bg-slate-700 rounded col-span-1"></div>
+                  <div class="h-2 bg-slate-700 rounded col-span-1"></div>oud
```
Comment (AydinHassan) [inline assets/components/ExerciseEditor.vue]:
```suggestion
                  <div class="h-2 bg-slate-700 rounded col-span-1"></div>
```

File: assets/components/ExerciseEditor.vue [hunk 6]
```diff
@@ -273,6 +284,9 @@ export default {
       </div>
     </div>
 
+    <div>
+    </div>
+
```
Comment (AydinHassan) [inline assets/components/ExerciseEditor.vue]:
```suggestion
```

File: assets/components/StudentProgress.vue [hunk 7]
```diff
@@ -64,4 +64,6 @@ export default {
       <div class="bg-pink-600 h-2.5 rounded-full dark:bg-pink-500" :style="{ 'width': percentComplete + '%' }"></div>
     </div>
   </div>
+
+
```
Comment (AydinHassan) [inline assets/components/StudentProgress.vue]:
```suggestion
```

**Review by AydinHassan — APPROVED** [review]
Looks good man, nice one! There are a few changes that look like they are accidents so I'll just go ahead and resolve those and get this merged!

**Comment by roryhmartin** [comment]
Sorry about those man, I should have checked over it. glad you liked the result though!

**Pull request merged**